lib/sles4sap/sap_deployment_automation_framework/configure_workload_tfvars.pm

```diff
@@ -0,0 +1,393 @@
+        # Defines if a custom dns solution is used
+        use_custom_dns_a_registration => 'false',
+        # Defines if the Virtual network for the Virtual machines is registered with DNS
+        # This also controls the creation of DNS entries for the load balancers
+        register_virtual_network_to_dns => 'true',
+        # Boolean value indicating if storage accounts and key vaults should be registered to the corresponding dns zones
+        register_storage_accounts_keyvaults_with_dns => 'false',
+        # If defined provides the DNS label for the Virtual Network
+        dns_label => '"openqa.net"'
+    );
+    return \%result;
+}
+
+=head2 define_nat_section
+
+    define_nat_section(environment=>'LAB', sdaf_region=>'swedencentral', workload_vnet_code=>'OpenQA-42');
+
+Returns tfvars section related to NAT setup in B<HASHREF> format.
+
+=over
+
+=item * B<environment>: SDAF environment
+
+=item * B<sdaf_region>: Public cloud location
+
+=item * B<workload_vnet_code>: Public cloud location
+
+=back
+
+=cut
+
+sub define_nat_section {
+    my (%args) = @_;
+    for my $arg ('environment', 'sdaf_region', 'workload_vnet_code') {
+        croak "Missing mandatory argument \$args{$arg}" unless $args{$arg};
+    }
+    my %result = (
+        header => '###  NAT Configuration ###',
+        deploy_nat_gateway => 'true',
+        nat_gateway_name => "\"$args{environment}-$args{sdaf_region}-$args{workload_vnet_code}-NG_0001\""
+    );
+
+    return (\%result);
+}
+
+=head2 define_subnets
+
+    define_subnets(network_data=>$network_data);
+
+Returns tfvars section related to subnet setup in B<HASHREF> format.
+
+=over
+
+=item * B<network_data>: Network data obtained from `lib/sap_deployment_automation_framework/networking::calculate_subnets`
+
+=back
+
+=cut
+
+sub define_subnets {
+    my (%args) = @_;
+    croak 'Missing mandatory argument $args{network_data}' unless $args{network_data};
+    my %result = (
+        header => '###  Subnet definitions ###',
+        network_address_space => "\"$args{network_data}->{network_address_space}\"",
+        iscsi_subnet_address_prefix => "\"$args{network_data}->{iscsi_subnet_address_prefix}\"",
+        web_subnet_address_prefix => "\"$args{network_data}->{web_subnet_address_prefix}\"",
+        admin_subnet_address_prefix => "\"$args{network_data}->{admin_subnet_address_prefix}\"",
+        db_subnet_address_prefix => "\"$args{network_data}->{db_subnet_address_prefix}\"",
+        app_subnet_address_prefix => "\"$args{network_data}->{app_subnet_address_prefix}\""
+    );
+
+    return (\%result);
+}
+
+=head2 define_iscsi_devices
+
+    define_iscsi_devices();
+
+Returns tfvars section related to setup of iSCSI server(s) in B<HASHREF> format.
+iSCSI is used mostly for SBD based fencing.
+
+=cut
+
+sub define_iscsi_devices {
+    # Convert OpenQA setting "SDAF_FENCING_MECHANISM" to value accepted by SDAF
+    my $fencing_type = get_fencing_mechanism();
```

Comment:
Maybe provide fencing_type as input argument

Reply:
I'd rather not do that as fencing type should be only set once either in job group or cmd, not midway during the test. 
That's the point of  `get_fencing_mechanism`.  Call the function, it will get you the data and checks it as well.